include/cooking/cookingJavascript/groceryListFormating.php

```diff
@@ -0,0 +1,38 @@
+<script>
+function newGroceryItem(){
+    var ingredientName = $('#ingredientName').val();
+    var addItem = $('#addItem').val();
+
+    if(ingredientName == ''){
+        console.log("im here 2");
+        return false;
+        alert("Please enter an ingredient name");
+        return false;
+    }
+    //console.log("we're here 2", addItem, ingredient);
+
+    $.post('/ajax/saveShoppingList.php', {ingredientName:ingredientName, addItem:addItem}, function(Data){
+          if (Data == "failure"){
+            alert("Error");
+            return false;
+        }
+        $('.allIngredients').html(Data)
+    });
+    return false;
+}
+// function strikeThrough(){
+//     var ingredient = $('#ingredientName').val();
+//     $('ingredient').strike();
+// }
+function deleteIngredient(ingredientId){
+    console.log(ingredientId);
+    var deleteBtn = $('#deleteBtn').val();
+    $.post('/ajax/deleteIngredientOffShoppingList.php', { deleteBtn:deleteBtn, ingredientId:ingredientId}, function(Data){
+        $('.allIngredients').html(Data)});
+        console.log("2");
+    return false;
+}
```

Comment:
This all looks good. I don't have much to add except that it looks like the UI doesn't do anything to indicate it's loading, so there's a moment between when you click the button and when the action has taken hold. I personally find that frustrating as a user, so I'd suggest adding a loading overlay or something like that. It might seem instant on your local machine, but it'll have more latency on the live site.

For bonus points, you could do an optimistic write by using js to modify the DOM to the action appear immediately even before the server has returned.